tests/models/test_taskinstance.py

```diff
@@ -1683,6 +1683,21 @@ def test_handle_failure(self, create_dummy_dag, session=None):
         assert context_arg_3 and "task_instance" in context_arg_3
         mock_on_retry_3.assert_not_called()
 
+    def test_handle_failure_updates_queued_task_try_number(self, dag_maker):
+        session = settings.Session()
+        with dag_maker():
+            task = DummyOperator(task_id="mytask", retries=1)
+        dr = dag_maker.create_dagrun()
+        ti = TI(task=task, run_id=dr.run_id)
+        ti.state = State.QUEUED
+        session.merge(ti)
+        session.commit()
+        assert ti.state == State.QUEUED
+        assert ti.try_number == 1
+        ti.handle_failure("test queued ti", test_mode=True)
+        assert ti.state == State.UP_FOR_RETRY
+        assert ti.try_number == 3
```

Comment:
Hmmm, this is right but my does it confusing!

Reply:
Quiet confusing. I have put breakpoints to see how it's happening but no result. If I remove the increment of the try_numer, the try_number remains 1 after running the handle_failure

Reply:
Yeah - `try_number` get's bumped to 2 in https://github.com/apache/airflow/blob/550ca654cd1b6c2e97a4174c3950a17093747880/airflow/models/taskinstance.py#L1722-L1723

and when `ti.try_number` is accessed here: https://github.com/apache/airflow/blob/550ca654cd1b6c2e97a4174c3950a17093747880/airflow/models/taskinstance.py#L467-L480

since it is not running and in  `UP_FOR_RETRY` state, the `try_number` is set to `3`.

Reply:
This might be a case where accessing `ti._try_number` would be less confusing.